### PR TITLE
Jedis 4 instrumentation

### PR DESCRIPTION
**Pull request by laurit**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/4554

### CODE REVIEW

**Review by trask — APPROVED** [review]
thx

**Comment by trask** [comment]
Merging early with just one approval to fix the build.

File: instrumentation/jedis/jedis-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v4_0/JedisConnectionInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        this.getClass().getName() + "$SendCommand2Advice");
+
+    transformer.applyAdviceToMethod(
+        isMethod()
+            .and(named("sendCommand"))
+            .and(takesArguments(2))
+            .and(takesArgument(0, named("redis.clients.jedis.commands.ProtocolCommand")))
+            .and(takesArgument(1, is(byte[][].class))),
+        this.getClass().getName() + "$SendCommandAdvice");
+    // FIXME: This instrumentation only incorporates sending the command, not processing the result.
```
Comment (iNikem) [inline instrumentation/jedis/jedis-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v4_0/JedisConnectionInstrumentation.java:50]: @laurit Please create an issue about this gap

Comment (laurit) [reply instrumentation/jedis/jedis-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jedis/v4_0/JedisConnectionInstrumentation.java:50]: @iNikem this is a copy paste from previous version

**Pull request merged**